### DISCUSSION TITLE
InsightFace descriptor differences

### DISCUSSION THREAD

**Discussion by hashimkhanzada** [vladmandic/human#428] [Q&A] [answered]
Hi,

I switched the default model to insightface. The first difference I noticed was the desciptor was being stored as Array(512) instead of the usual Array(1024).

1. One of the threads mentioned fewer elements in a descriptor means less accuracy, but how would that work in this case where insightface is meant to be more accurate than the default.

2. I'm assuming there's no way to make my existing descriptors (made by the default model, 1024 element descriptors) compatible with insightface, and I'd have to recalculate them all?

3. the similarity returned by insightface was much lower than faceres, on the same images. If i took a screenshot of my face, faceres would be >80% whereas insightface would barely reach 20%. Is this intentional and meant to be dealt with by playing around with the matchOptions (multiplier, min, max)? 

Thanks for the great work on this library though. Feels much more comprehensive than the original face-api.js

**Comment by vladmandic** [comment, marked as answer]
- different feature vectors are not cross-compatible.
- scores are not in the same range, that's exactly why multiplier/min/max exist
- different feature vector models are tuned for different conditions and are highly sensitive to items such as face-crop factor.

all those have been covered in previous conversations in details, so you can search for more info.
